src/cdk/a11y/key-manager/activedescendant-key-manager.ts

```diff
@@ -24,11 +24,22 @@ export interface Highlightable extends ListKeyManagerOption {
 export class ActiveDescendantKeyManager<T> extends ListKeyManager<Highlightable & T> {
 
   /**
-   * This method sets the active item to the item at the specified index.
-   * It also adds active styles to the newly active item and removes active
-   * styles from the previously active item.
+   * Sets the active item to the item at the specified index and adds the
+   * active styles to the newly active item. Also removes active styles
+   * from the previously active item.
+   * @param index Index of the item to be set as active.
    */
-  setActiveItem(index: number): void {
+  setActiveItem(index: number): void;
+
+  /**
+   * Sets the active item to the item to the specified one and adds the
+   * active styles to the it. Also removes active styles from the
+   * previously active item.
+   * @param item Item to be set as active.
+   */
+  setActiveItem(item: T): void;
```

Comment:
Technically `T` could be `number`. It would avoid the ambiguity if we just kept the separate methods (`setActiveItem` and `setActiveItemIndex`)

Reply:
I don't think so. The `_items` that are passed in through the constructor are a `QueryList` which can't contain numbers.

Reply:
Is there ever a time when a component would want to use one or the other in a single call? Still seems like it would be more explicit to have both methods with index vs. item behaviors.

Reply:
It would be more explicit, but it adds a couple of extra methods to the API that do more or less the same. I don't think there's a case where a component would want to use both of them in the same call.

Reply:
Ack, I don't feel that strongly. My only other comment would be that the implemention method should be `item: T|number`

Reply:
That's what I was going for initially, but then TS seems to have a hard time figuring out the type when we pass the item to `updateActiveItem`. There were also some issues when trying to assign the active item inside the derived classes. I went with `any` since the implementation method signature won't show up in the type info anyway.